Orange/widgets/data/owpreprocess.py

```diff
@@ -535,6 +536,125 @@ def createinstance(params):
 # TODO: Model based FS (random forest variable importance, ...), RFE
 # Unsupervised (min variance, constant, ...)??
 
+class RandomFeatureSelectEditor(BaseEditor):
+    #: Strategy
+    Fixed, Percentage, FDR, FPR, FWE = 0, 1, 2, 3, 4
+
+    def __init__(self):
+        super().__init__()
+        self._spins = {}
+        self._max_num_attributes = None
+        self._ticks_to_update_slider = [None, None]
+        self._strategy = RandomFeatureSelectEditor.Fixed
+        self._k = 10
+        self._p = 75
+
+        self.setLayout(QVBoxLayout())
+        box = gui.widgetBox(widget=self, box="Strategy", spacing=8)
+        strategy_buttons = gui.radioButtons(widget=box, master=self,
+                                            value="_strategy",
+                                            callback=self.update_strategy)
+
+        fixed_rb = gui.appendRadioButton(group=strategy_buttons, label="Fixed:",
+                                         insertInto=box)
+        fixed_rb.setMinimumHeight(25)
+        ibox = gui.indentedBox(widget=fixed_rb, sep=63)
+        k_spin = gui.spin(
+            widget=ibox, master=self, value="_k", minv=1, maxv=10, step=1,
+            label="", controlWidth=60, spinType=int, callback=self.edited.emit,
+            disabled=self._strategy != RandomFeatureSelectEditor.Fixed)
+        self._spins[RandomFeatureSelectEditor.Fixed] = k_spin
+
+        percentage_rb = gui.appendRadioButton(group=strategy_buttons,
+                                              label="Percentage:",
+                                              insertInto=box)
+        ibox = gui.indentedBox(widget=box)
+        p_spin = gui.hSlider(
+            widget=ibox, master=self, value="_p", minValue=0, maxValue=100,
+            step=4, ticks=20, labelFormat="%d %%", callback=self.update_p_spin,
+            disabled=self._strategy != RandomFeatureSelectEditor.Percentage)
+        self._spins[RandomFeatureSelectEditor.Percentage] = p_spin
```

Comment:
@matevzkren, with this not working out of the box, I don't think there is much reason to obfuscate code by using gui.py widgets. So either have this class extend `gui.OWComponent` (some adjustment might be necessary) to reap its benefits (i.e. dropping the setter methods as noted in the comment above), or revert to using Qt directly as the other widgets here do (in which case I'm sorry to have misled you :smile:).